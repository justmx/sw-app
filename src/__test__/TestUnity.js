import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

export const ReduxProvider = ({ children, initialState = {} }) => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore(initialState);
  return <Provider store={store}>{children}</Provider>;
};
