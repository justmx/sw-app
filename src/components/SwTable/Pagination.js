import React from "react";

const Pagination = ({ nextPage, previouPage }) => {
  return (
    <div className="pagination">
      <span className="pagination-title">Pagination</span>
      <span className="pagination-button" onClick={previouPage}>
        {"< Back"}
      </span>
      <span className="pagination-button" onClick={nextPage}>
        {"Next >"}
      </span>
    </div>
  );
};

export default Pagination;
