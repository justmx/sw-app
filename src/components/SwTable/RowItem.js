import React from "react";

const RowItem = ({ person, onClick }) => {
  if (!person) return null;
  const { name, height, mass } = person;

  return (
    <div className="row" onClick={onClick}>
      <div className="cell">{name}</div>
      <div className="cell">{height}</div>
      <div className="cell">{mass}</div>
    </div>
  );
};

export default RowItem;
