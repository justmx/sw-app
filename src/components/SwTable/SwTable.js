import React from "react";
import debounce from "lodash.debounce";
import { useSelector, useDispatch } from "react-redux";
import { getPersonDetail } from "../../redux/sw-reducer";
import Pagination from "./Pagination";
import RowItem from "./RowItem";

const SwTableSection = ({ nextPage, previouPage }) => {
  const dispatch = useDispatch();
  const people = useSelector((state) => state.swReducer.people);

  const onClick = debounce((person) => {
    dispatch(getPersonDetail(person));
  }, 150);

  return (
    <div className="sw-table-section">
      <h1>Table with list of people</h1>
      <div className="swap-table-container">
        <div className="table">
          <div className="row header blue">
            <div className="cell">Name</div>
            <div className="cell">Height</div>
            <div className="cell">Mass</div>
          </div>
          {people.map((person, i) => {
            return (
              <RowItem
                key={i}
                person={person}
                onClick={() => onClick(person)}
              />
            );
          })}
        </div>
        {people.length > 0 && (
          <Pagination nextPage={nextPage} previouPage={previouPage} />
        )}
      </div>
    </div>
  );
};

export default SwTableSection;
