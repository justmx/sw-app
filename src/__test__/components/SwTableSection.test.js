import React from "react";
import { shallow, mount } from "enzyme";
import { ReduxProvider } from "../TestUnity";
import SwTable from "../../components/SwTable/SwTable";
import RowItem from "../../components/SwTable/RowItem";
import Pagination from "../../components/SwTable/Pagination";

const people = [
  {
    name: "Beru Whitesun lars",
    height: "165",
    mass: "136",
  },
  {
    name: "C-3PO",
    height: "167",
    mass: "123",
  },
];

const selectedPerson = {
  name: "Beru Whitesun lars",
  height: "165",
  mass: "136",
  allfilms: [
    { title: "The Empire Strikes Back" },
    { title: "Return of the Jedi" },
  ],
};

const initialState = {
  swReducer: {
    people,
    selectedPerson,
  },
};

describe("Render SwTable", () => {
  const wrapper = shallow(
    <ReduxProvider>
      <SwTable />
    </ReduxProvider>
  );

  it("should render SwTable", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Check SwTable Childcomponent", () => {
  const wrapper = mount(
    <ReduxProvider initialState={initialState}>
      <SwTable />
    </ReduxProvider>
  );
  it("should render child components", () => {
    expect(wrapper.find(RowItem)).toHaveLength(2);
    expect(wrapper.find(Pagination)).toHaveLength(1);
  });
});
