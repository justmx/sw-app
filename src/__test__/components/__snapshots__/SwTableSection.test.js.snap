// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SwTable should render SwTable 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SwTableSection />
</Provider>
`;
